src/hook/useProducts.ts

```diff
@@ -0,0 +1,70 @@
+import { useState, useEffect } from 'react';
+import { Product } from '../types/product';
+import { ProductAPIService } from '../services/ProductAPIService';
+import { PRODUCTS_MOCK as mockProducts } from '../mock/products';
+
+const productsPerPage = 8;
+const useMockData = import.meta.env.USE_MOCK_FOR_API_FAIL === 'true';
+
+interface UseProductsResult {
+  products: Product[];
+  isLoading: boolean;
+  hasMore: boolean;
+  loadMore: () => void;
+}
+
+const fetchProducts = async (category: string, page: number) => {
+  if (!category) {
+    return { data: [], total: 0 };
+  }
+
+  if (!useMockData) {
```

Comment:
Do we need this? if yes, revert the condition check if `useMockData === true` then we fast return mock value, so that we remove 1 nested level of bunch logic below

Reply:
Fixed